src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java

```diff
@@ -102,19 +100,28 @@ public void testNonResolvingIndexPatternWithNonExistingIndex() throws IOExceptio
     }
 
     @Test
-    public void testNonIdenticalMappings() throws IOException {
+    public void testNonCompatibleMappings() throws IOException {
         try {
-            explainQuery(String.format(Locale.ROOT, "SELECT firstname, birthdate FROM %s, %s " +
-                            "WHERE firstname = 'Leo' OR male = 'true'",
-                    TEST_INDEX_BANK, TEST_INDEX_ONLINE));
+            explainQuery(String.format(Locale.ROOT, "SELECT * FROM %s, %s ",
```

Comment:
please add `Assert.fail()` after this line (in the `try` block), otherwise if no exception is thrown your test will still pass.

Reply:
Good point. Added that as you pointed out.